util/src/test/java/tech/pegasys/artemis/util/backing/ContainerViewTest.java

```diff
@@ -0,0 +1,365 @@
+      Assertions.assertEquals(UnsignedLong.ZERO, c1w.getList3().get(0).getLong1());
+      Assertions.assertEquals(Bytes32.ZERO, c1w.getList3().get(0).getBytes1());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x999), c1w.getList3().get(1).getLong1());
+      Assertions.assertEquals(
+          Bytes32.leftPad(Bytes.fromHexString("0xa999")), c1w.getList3().get(1).getBytes1());
+    }
+
+    ContainerRead c1r = c1w.commitChanges();
+    TreeUtil.dumpBinaryTree(c1r.getBackingNode());
+
+    {
+      Assertions.assertEquals(UnsignedLong.ZERO, c1.getSub1().getLong1());
+      Assertions.assertTrue(c1.getList1().isEmpty());
+      Assertions.assertTrue(c1.getList2().isEmpty());
+      Assertions.assertEquals(UnsignedLong.ZERO, c1.getList3().get(0).getLong1());
+      Assertions.assertEquals(Bytes32.ZERO, c1.getList3().get(0).getBytes1());
+      Assertions.assertEquals(UnsignedLong.ZERO, c1.getList3().get(1).getLong1());
+      Assertions.assertEquals(Bytes32.ZERO, c1.getList3().get(1).getBytes1());
+
+      Assertions.assertEquals(UnsignedLong.valueOf(0x1), c1r.getLong1());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x2), c1r.getLong2());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x111), c1r.getSub1().getLong1());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x222), c1r.getSub1().getLong2());
+      Assertions.assertEquals(2, c1r.getList1().size());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x333), c1r.getList1().get(0).get());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x444), c1r.getList1().get(1).get());
+      Assertions.assertThrows(
+          IndexOutOfBoundsException.class,
+          () -> {
+            c1r.getList1().get(2);
+          });
+      Assertions.assertEquals(2, c1r.getList2().size());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x555), c1r.getList2().get(0).getLong1());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x666), c1r.getList2().get(0).getLong2());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x777), c1r.getList2().get(1).getLong1());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x888), c1r.getList2().get(1).getLong2());
+      Assertions.assertEquals(UnsignedLong.ZERO, c1r.getList3().get(0).getLong1());
+      Assertions.assertEquals(Bytes32.ZERO, c1r.getList3().get(0).getBytes1());
+      Assertions.assertEquals(UnsignedLong.valueOf(0x999), c1r.getList3().get(1).getLong1());
+      Assertions.assertEquals(
+          Bytes32.leftPad(Bytes.fromHexString("0xa999")), c1r.getList3().get(1).getBytes1());
+    }
+
+    ContainerWrite c2w = c1r.createWritableCopy();
+    c2w.getList2().getByRef(1).setLong2(UnsignedLong.valueOf(0xaaa));
+    ContainerRead c2r = c2w.commitChanges();
+
+    Assertions.assertEquals(UnsignedLong.valueOf(0x888), c1r.getList2().get(1).getLong2());
+    Assertions.assertEquals(UnsignedLong.valueOf(0xaaa), c2r.getList2().get(1).getLong2());
```

Comment:
As a general standard, we usually use `assertJ` for test assertions:
```suggestion
    assertThat(c2r.getList2().get(1).getLong2()).isEqualTo(UnsignedLong.valueOf(0xaaa));
```

Reply:
Migrated tests.